HLTrigger/special/src/HLTL1NumberFilter.cc

```diff
@@ -76,6 +84,10 @@ HLTL1NumberFilter::filter(edm::StreamID, edm::Event& iEvent, const edm::EventSet
     if (data.data() && data.size() > 0){
       FEDHeader header(data.data()) ;
       if (period_!=0) accept = ( ( (header.lvl1ID())%period_ ) == 0 );
+      if (useTCDS_ && period_!=0) {
+	evf::evtn::TCDSRecord record((unsigned char *)data.data());
+	accept = ( (record.getHeader().getData().header.triggerCount % period_) == 0 );
+      }
```

Comment:
It may be cleaner to get a `counter` from either TCDS or the lvl1ID, depending on the configuration, and then write the `(counter % period_) == 0` logic only once.